kaldb/src/main/java/com/slack/kaldb/blobfs/s3/S3BlobFs.java

```diff
@@ -227,7 +228,7 @@ private boolean copyFile(URI srcUri, URI dstUri) throws IOException {
   public boolean mkdir(URI uri) throws IOException {
     LOG.info("mkdir {}", uri);
     try {
-      Preconditions.checkNotNull(uri, "uri is null");
+      Objects.requireNonNull(uri, "uri is null");
```

Comment:
Per guava [documentation](https://github.com/google/guava/wiki/PreconditionsExplained) it seems Preconditions.checkNotNull is still recommended. Do we still want to ban it?

Reply:
Hmm! Not sure? We use both with just one message argument.

Anyways if you think this rule is unnecessary I can remove this rule and back out the change

Reply:
Per the guava docs:
> Simple, varargs "printf-style" exception messages. (This advantage is also why we recommend continuing to use checkNotNull over Objects.requireNonNull)

Reply:
Let's stick with the Preconditions API for now. Also, ban Objects.requireNotNull if it's easy to do.